test/lib/rules/consistent-component-name.js

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+'use strict';
+
+const { RuleTester } = require('eslint');
+
+const { ESLINT_TEST_CONFIG } = require('../shared');
+const rule = require('../../../lib/rules/consistent-component-name');
+
+const ruleTester = new RuleTester(ESLINT_TEST_CONFIG);
+
+ruleTester.run('consistent-class-name', rule, {
+    valid: [
+        {
+            code: `export class Foo extends LightningElement {}`,
+            filename: 'foo.js',
+        },
+        {
+            code: `export class ComplexName extends LightningElement {}`,
+            filename: 'complexName.js',
+        },
+    ],
+    invalid: [
```

Comment:
How about a test case where the casing in the class name is off but the characters are correct?
```
code: `export class complexName extends LightningElement {}`,
filename: 'complexName.js'
```